### PR TITLE
Finish week 1 exercises

### PR DESCRIPTION
**Pull request by DaSpood**
Need review :cowboy_hat_face: @SanderJSA @CohenArthur @Skallwar @n1tram1

### CODE REVIEW

File: 0_intro/exercises/bubble_sort/src/lib.rs [hunk 1]
```diff
@@ -1,5 +1,13 @@
 pub fn bubble_sort(v: &mut Vec<i32>) {
-    todo!()
+    for i in 0..v.len() {
+        for j in 0..(v.len() - i - 1) {
+            if v[j] > v[j + 1] {
+                let tmp = v[j];
+                v[j] = v[j + 1];
+                v[j + 1] = tmp;
```
Comment (SanderJSA) [inline 0_intro/exercises/bubble_sort/src/lib.rs]: This can be simplified with `v.swap(j, j + 1);`

File: 0_intro/exercises/factorial/src/lib.rs [hunk 2]
```diff
@@ -1,5 +1,9 @@
 pub fn factorial(n: i32) -> i32 {
-    todo!()
+    match n {
+        0 => 1,
+        1 => 1,
```
Comment (SanderJSA) [inline 0_intro/exercises/factorial/src/lib.rs]: Perfectly valid but can also be written as `0 | 1 => 1,`

File: 0_intro/exercises/hello_world/src/main.rs [hunk 3]
```diff
@@ -1,4 +1,3 @@
 fn main() {
-    todo!("Print hello world on stdout! (not stderr)")
-    // There is no trap it really is a simple exercice
+    println!("hello world")
```
Comment (SanderJSA) [inline 0_intro/exercises/hello_world/src/main.rs:2]: trop fort

File: 0_intro/exercises/vec_sum/src/lib.rs [hunk 4]
```diff
@@ -1,5 +1,5 @@
 pub fn vec_sum(vec: Vec<i32>) -> i32 {
-    todo!()
+    vec.iter().fold(0, |acc, elt| { acc + elt }) // or .sum()
```
Comment (SanderJSA) [inline 0_intro/exercises/vec_sum/src/lib.rs]: Perfect !

**Review by SanderJSA — APPROVED** [review]
LGTM !

File: 0_intro/exercises/vec_strlen/src/lib.rs [hunk 5]
```diff
@@ -1,5 +1,7 @@
 pub fn vec_strlen(vec: Vec<&str>) -> Vec<usize> {
-    todo!()
+    vec.iter()
+       .map(|elt| { elt.len() })
```
Comment (CohenArthur) [inline 0_intro/exercises/vec_strlen/src/lib.rs]: If it's a simple lambda, you don't need the curly braces you Java boy (but it's still completely valid)
```suggestion
       .map(|elt| elt.len())
```

File: 0_intro/exercises/vec_sum/src/lib.rs (hunk 4 above)
Comment (CohenArthur) [inline 0_intro/exercises/vec_sum/src/lib.rs]: Same as above

**Review by CohenArthur — APPROVED** [review]
Good job :D

**Review by SanderJSA — APPROVED** [review]

**Pull request merged**